### PR TITLE
FUSETOOLS-3269 - Improve error message when incompatible Maven used with

### PR DESCRIPTION
**Pull request by apupier**
Syndesis

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [x] Are there integration tests (or at least a jira to tackle these)?
- [x] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
previous error message

```
[INFO] --- extension-maven-plugin:1.6.13.fuse-731002-redhat-00001:repackage-extension (default) @ custom-step-camel ---
[INFO] Resolving artifact io.syndesis.extension:extension-bom:pom:1.6.13.fuse-731002-redhat-00001 from [apache.snapshots (https://repository.apache.org/content/repositories/snapshots/, default, snapshots), redhat-ga (https://maven.repository.redhat.com/ga, default, releases), central (https://repo.maven.apache.org/maven2, default, releases)]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14.260 s
[INFO] Finished at: 2019-07-22T11:36:21+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.syndesis.extension:extension-maven-plugin:1.6.13.fuse-731002-redhat-00001:repackage-extension (default) on project custom-step-camel: Execution default of goal io.syndesis.extension:extension-maven-plugin:1.6.13.fuse-731002-redhat-00001:repackage-extension failed: Unable to invoke onlyOne([Ljava.lang.Class;@6af609ea) on object org.jboss.shrinkwrap.resolver.spi.loader.ServiceRegistry with parameters [Ljava.lang.Object;@280fafd: InvocationTargetException: Could not create new service instance: org/eclipse/aether/internal/impl/DefaultDependencyCollector: org.eclipse.aether.internal.impl.DefaultDependencyCollector -> [Help 1]
```


new error message:

```
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven) @ custom-step-camel ---
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.273 s
[INFO] Finished at: 2019-07-22T11:30:08+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven) on project custom-step-camel: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
```

notice also that the error message is appearing 10 times faster

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml [hunk 1]
```diff
@@ -68,10 +69,31 @@
...
+					<execution>
+						<id>enforce-maven</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireMavenVersion>
+									<version>[3.5,3.6)</version>
+									<message>Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.</message>
```
Comment (bfitzpat) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml]: The initial message text seems a little repetitive to me. I would change the message slightly from: "Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

to: "The Maven version must be a 3.5.x version in Eclipse to work correctly. If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

Also, I am wondering if this is this a temporary problem and we'll be getting rid of the JIRA link eventually?

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/step/camelroute/pom.xml [hunk 2]
```diff
@@ -74,6 +74,27 @@
...
+					<execution>
+						<id>enforce-maven</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireMavenVersion>
+									<version>[3.5,3.6)</version>
+									<message>Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.</message>
```
Comment (bfitzpat) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/step/camelroute/pom.xml]: See above, but I would change this message to: "The Maven version must be a 3.5.x version in Eclipse to work correctly. If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/step/javabean/pom.xml [hunk 3]
```diff
@@ -93,6 +93,27 @@
...
+					<execution>
+						<id>enforce-maven</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireMavenVersion>
+									<version>[3.5,3.6)</version>
+									<message>Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.</message>
```
Comment (bfitzpat) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/step/javabean/pom.xml]: See above, but I would change this message to: "The Maven version must be a 3.5.x version in Eclipse to work correctly. If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml (hunk 1 above)
Comment (apupier) [reply syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml]: > Also, I am wondering if this is this a temporary problem and we'll be getting rid of the JIRA link eventually?

very low probability that we will get rid of it on short term given the amount of time allocated to investigate on it and that several people already had a look.
We will get rid of it when all dependencies will be fixed and upgraded. (Maybe Fuse 8?) Unfortunately again on this point, seems the root project is quite abandoned without a lot of maintenance resource allocated.

Comment (apupier) [reply syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml]: > The Maven version must be a 3.5.x version in Eclipse to work correctly.

it is not specifically "in Eclipse" that it needs to be 3.5.x, it is for all Maven builds of Fuse Online extension.

> If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.

re-reading, the part I initially provided and yours makes me realize we could be more precise for this sentence on the "Maven in Eclipse".

With both previous comments, what do you think of the following proposal?

"The Maven version must be a 3.5.x version to work correctly. If using the embedded Maven in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

Comment (bfitzpat) [reply syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml]: Yes! I agree with that last revision.

**Pull request merged**